projects/storefrontlib/src/cms-components/misc/qualtrics/qualtrics-loader.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { BehaviorSubject, fromEvent, Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, mergeMap, tap } from 'rxjs/operators';
+import { QualtricsConfig } from '../../../shared/config/qualtrics-config';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QualtricsLoaderService {
+  qualtricsLoaded$ = new BehaviorSubject<boolean>(false);
+
+  constructor(private winRef: WindowRef, private config: QualtricsConfig) {
+    if (Boolean(this.winRef.nativeWindow)) {
+      this.initialize();
+      this.setup();
+    }
+  }
+
+  load(): Observable<boolean> {
+    return this.qualtricsLoaded$.pipe(
+      filter(loaded => loaded),
+      mergeMap(_ => {
+        const qsi = this.winRef.nativeWindow['QSI'];
+        return this.isDataLoaded().pipe(
+          distinctUntilChanged(),
+          tap(dataLoaded => {
+            if (dataLoaded) {
+              qsi.API.unload();
+              qsi.API.load().done(qsi.API.run());
+            }
+          })
+        );
+      })
+    );
+  }
+
+  private isQualtricsConfigured(): boolean {
+    return (
+      Boolean(this.config.qualtrics) && Boolean(this.config.qualtrics.projectId)
+    );
+  }
+
+  private initialize() {
```

Comment:
Please move up, to be follow the chronological order of the call stack - in the constructor, the first thing that's called is this `initialize()`, and then `setup()`, etc.